backbone/src/main/java/org/researchstack/backbone/ui/task/TaskViewModel.kt

```diff
@@ -149,10 +146,13 @@ internal class TaskViewModel(context: Application, intent: Intent) : AndroidView
     fun previousStep() {
         Log.d(TAG, "1. CURRENT STEP: $currentStep")
         if (editing) {
-            currentStep = stack.pop()
-            currentStepEvent.value = StepNavigationEvent(step = currentStep!!, isMovingForward = false)
-
-            editing = stack.isEmpty().not()
+            val current = stack.pop()
+            if (!stack.isEmpty().not()) {
```

Comment:
Is this correct?

`!stack.isEmpty().not()` is a double negation... is that what you meant?